speechbrain/utils/seed.py

```diff
@@ -0,0 +1,73 @@
+"""Seed utilities for reproducibility.
+
+Authors
+ * Adel Moumen 2024
+"""
+
+import logging
+import os
+import random
+
+import torch
+
+from speechbrain.utils.distributed import get_rank, rank_prefixed_message
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+max_seed_value = 4294967295  # 2^32 - 1 (uint32)
+min_seed_value = 0
+
+
+def seed_everything(
+    seed: int = 0, verbose: bool = True, deterministic: bool = False
+) -> int:
+    r"""Function that sets the seed for pseudo-random number generators in: torch, numpy, and Python's random module.
+
+    Arguments
+    ---------
+    seed: the integer value seed for global random state.
+    verbose: Whether to print a message on each rank with the seed being set.
+    deterministic: Whether to set the seed for deterministic operations.
+
```

Comment:
This docstring is not in the right format, should be:

```
variable: type
  description
```

Reply:
Done.